robocop/checkers/misc.py

```diff
@@ -258,3 +258,52 @@ def visit_LibraryImport(self, node):  # noqa
         if not node.name:
             return
         self.libraries.append(node)
+
+
+class UnusedVariableChecker(VisitorChecker):
+    """ Checker for unused variable. """
+    rules = {
+        "0912": (
+            "variable-not-used",
+            "The variable is assigned but not used in this scope",
+            RuleSeverity.WARNING
+        )
+    }
+
+    def __init__(self):
+        self.variable_list = []
+        super().__init__()
+
+    def visit_KeywordCall(self, node):  # noqa
+        for child in node.data_tokens:
```

Comment:
You can assign variables with the same name as your arguments:
```
${var} Keyword ${var} 
```
And variable after redeinition would be not used. Thats why you should first get only arguments (and save assign to temporary list for now), call use_variaes on it,  and then process assign.

variable_list should also pop/remove variables when using them - so you will catch redefined but not used vars